### ISSUE TITLE
`type-annotation-spacing`:  Inappropriate warning about `overrides`

### ISSUE THREAD

**Issue by finscn** [eslint-stylistic/eslint-stylistic#1079]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

when I use 
```json
      "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": false,
          "after": true,
          "overrides": {
            "arrow": {
              "before": true,
              "after": true
            }
          }
        }
      ],
```

the warning is 
```
[@stylistic/eslint-plugin]: You are using deprecated options("overrides.arrow") in "type-annotation-spacing", please use "arrow-spacing" instead.
```

But if I change the config with `arrow-spacing` , it can't work at all.
And as the document says ( https://eslint.style/rules/type-annotation-spacing#type-annotation-spacing ) 
the `overrides.arrow` is an right option .

### Reproduction

https://stackblitz.com/edit/github-dehiwzsd?file=eslint.config.js

### System Info

```shell
macos
the package dependencies : 

    "@stylistic/eslint-plugin": "^5.6.1",
    "@stylistic/eslint-plugin-migrate": "^4.4.1",
    "@types/node": "^25.0.2",
    "@typescript-eslint/parser": "^8.50.0",
    "globals": "^16.5.0",
    "typescript-eslint": "^8.50.0"
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
> But if I change the config with `arrow-spacing` , it can't work at all.

Please provide a minimal reproduction.

And the `override.arrow` option is marked as deprecated in the [documentation](https://eslint.style/rules/type-annotation-spacing#arrow-deprecated).

**Comment by finscn** [comment]
the document doesn't tell me  how to use  `arrow-spacing` in `type-annotation-spacing`

```json
      "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": false,
          "after": true,
          "overrides": {
            "arrow-spacing": {
              "before": true,
              "after": true
            }
          }
        }
      ],
```
is wrong
```
Error: Key "rules": Key "stylistic/type-annotation-spacing":
        Value {"arrow-spacing":{"before":true,"after":true}} should NOT have additional properties.
                Unexpected property "arrow-spacing". Expected properties: "colon", "arrow", "variable", "parameter", "property", "returnType".
```

------------
And
```json
      "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": false,
          "after": true,
          "arrow-spacing": {
            "before": true,
            "after": true
          }
        }
      ],
```
is  also  wrong
```
Error: Key "rules": Key "stylistic/type-annotation-spacing":
        Value {"before":false,"after":true,"arrow-spacing":{"before":true,"after":true}} should NOT have additional properties.
                Unexpected property "arrow-spacing". Expected properties: "before", "after", "overrides".
```

------

How to understand  `please use "arrow-spacing" instead.`  ?

**Comment by 9romise** [comment]
`arrow-spacing` is a rule, see: https://eslint.style/rules/arrow-spacing. You can search the name in the documention for more details.

I'll try to improve the warning message.

**Comment by finscn** [comment]
I use 

Code A : 
```json
      "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": false,
          "after": true
        }
      ],
      "stylistic/arrow-spacing": [
        "error",
        {
          "before": true,
          "after": true
        }
      ],
```

 But the effect does not equal  `overrides.arrow` : 
Code B
```json
      "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": false,
          "after": true,
          "overrides": {
            "arrow": {
              "before": true,
              "after": true
            }
          }
        }
      ]
```

the `Code A` doesn't allow the space around `=>' .

the is an error in vscode

<img width="513" height="36" alt="Image" src="https://github.com/user-attachments/assets/1ffd8b12-f603-45a1-beaf-39ef5fb658fc" />

<img width="729" height="32" alt="Image" src="https://github.com/user-attachments/assets/8d475d11-327e-4b11-a5f8-91c21836229e" />

**Comment by finscn** [comment]
I found the answer , I have to use this: 
```
      "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": true,
          "after": true,
          "overrides": {
            "colon": {
              "before": false,
              "after": true
            }
          }
        }
      ],
      "stylistic/arrow-spacing": [
        "error",
        {
          "before": true,
          "after": true
        }
      ],
```

change  `type-annotation-spacing` to  true & true , then use `overrides.colon` ...

the setting of `type-annotation-spacing`  have to be the same as `arrow-spacing`, 
otherwise `arrow-spacing` can't work corretly

I think this is a bad design.

**Comment by 9romise** [comment]
In fact, your settings are the same as the default values, so you could simply keep the defaults to achieve the same effect.
```
"stylistic/type-annotation-spacing": ["error"],
"stylistic/arrow-spacing": ["error"],
```
Or if you want to modify `colon` only,
```
"stylistic/type-annotation-spacing": ["error", {
	"override": {
		"colon": {
			"before": true,
			"after": true,
		}
	}
}],
"stylistic/arrow-spacing": ["error"],
```
The `arrow` is `{ "before": true, "after": true }` by default, which is specified in the [document](https://eslint.style/rules/type-annotation-spacing).

**Comment by 9romise** [comment]
Well, this is not by design. Starting from version 5.7.0, you can use:
```
     "stylistic/type-annotation-spacing": [
        "error",
        {
          "before": false,
          "after": true,
          "overrides": {
            "arrow": "ignore"
          }
        }
      ],
      "stylistic/arrow-spacing": [
        "error",
        {
          "before": true,
          "after": true
        }
      ],
```
To avoid breaking changes, this is a trade-off we're making for now. We will explore ways to improve such modifications in future releases.
